crates/iota-sdk-types/src/lib.rs

```diff
@@ -176,30 +177,62 @@ pub use transaction::{
 };
 pub use type_tag::{Identifier, StructTag, TypeParseError, TypeTag};
 
-#[cfg(feature = "uniffi")]
-uniffi::setup_scaffolding!();
-
-#[cfg(test)]
+#[cfg(all(test, feature = "serde", feature = "proptest"))]
 mod serialization_proptests;
 
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct PersonalMessage<'a>(pub std::borrow::Cow<'a, [u8]>);
 
-#[cfg(feature = "uniffi")]
 #[macro_export]
-macro_rules! impl_uniffi_byte_vec_wrapper {
-    ($id:ident) => {
-        uniffi::custom_type!($id, Vec<u8>, {
-            lower: |val| val.as_bytes().to_vec(),
-            try_lift: |vec| Ok($id::from_bytes(vec)?),
-        });
+macro_rules! def_is {
+    ($($variant:ident),* $(,)?) => {
+        paste::paste! {$(
+        #[inline]
+        pub fn [< is_ $variant:snake >](&self) -> bool {
+            matches!(self, Self::$variant)
+        }
+        )*}
     };
 }
 
-#[cfg(not(feature = "uniffi"))]
 #[macro_export]
-macro_rules! impl_uniffi_byte_vec_wrapper {
-    ($id:ident) => {};
+macro_rules! def_is_as_opt {
+    ($($variant:ident => $inner:ty),* $(,)?) => {
+        paste::paste! {$(
+        #[inline]
+        pub fn [< is_ $variant:snake >](&self) -> bool {
+            matches!(self, Self::$variant(_))
+        }
+
+        #[inline]
+        pub fn [< as_ $variant:snake >](&self) -> &$inner {
+            let Self::$variant(inner) = self else {
+                panic!("not a {}", stringify!($variant));
+            };
+            inner
+        }
+
+        #[inline]
+        pub fn [< as_ $variant:snake _opt >](&self) -> Option<&$inner> {
+            if let Self::$variant(inner) = self {
+                Some(inner)
+            } else {
+                None
+            }
+        }
+
+        #[inline]
+        pub fn [< into_ $variant:snake >](self) -> $inner {
```

Comment:
Maybe this should be mentioned in the macro name

Reply:
I thought about it but does it really matter?

Reply:
Just thought it was weird and arbitrary that one method is not mentioned